presto-oracle/src/test/java/io/prestosql/plugin/oracle/TestOracleTypes.java

```diff
@@ -141,9 +141,9 @@ public void testBooleanType()
     @Test
     public void testSpecialNumberFormats()
     {
-        oracleServer.execute("CREATE TABLE test (num1 number, num2 number(*,-2))");
-        oracleServer.execute("INSERT INTO test VALUES (12345678901234567890.12345678901234567890123456789012345678, 1234567890.123)");
-        assertQuery("SELECT * FROM test", "VALUES (12345678901234567890.1234567890, 1234567900.0000000000)");
+        oracleServer.execute("CREATE TABLE test (num1 number)");
```

Comment:
the test of negative scale will be moved to another side?

Reply:
There are tests for negative scale in `testNumberNegativeScaleReadMapping`.
This specific case which was tested here is not supported as it would require decimal precision of 40 in Presto, while up to 38 is supported.

Reply:
really not, as -2 will affect the first part..
this 1234567890 will be saved as 1234567900

Reply:
The problem is that Oracle will allow for numbers with up to 38 **meaningful** digits for types with negative scale (in your example meaningful digits are 12345679).
The number of total digits (including obvious trailing zeroes) larger. 

E.g. for `NUMBER(*,-2) total number of digits in represented values can be 40.
If you run such query in Oracle: 
```
SELECT CAST(1234567890123456789012345678901234567890 AS NUMBER(*,-2)) FROM dual;
```
It will work fine and return 40 digit number `1234567890123456789012345678901234567900`.

As we do not support negative scale in Presto, we need to be able to represent the value, including trailing zeroes. Hence the limitation.